### PR TITLE
De-duplicate `DistributedSampler` mentions

### PR DESCRIPTION
**Pull request by carmocca**
## What does this PR do?

Noticed our docs were messy while answering https://github.com/PyTorchLightning/pytorch-lightning/discussions/7616

## Before submitting
- [ ] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [n/a] Did you make sure to update the documentation with your changes? (if necessary)
- [n/a] Did you write any new necessary tests? (not for typos and docs)
- [n/a] Did you verify new and existing tests pass locally with your changes?
- [n/a] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

### CODE REVIEW

File: docs/source/clouds/cluster.rst [hunk 1]
```diff
@@ -301,26 +301,3 @@ Set the ``NCCL_DEBUG=INFO`` environment variable to see the ACTUAL error.
 .. code-block:: bash
 
     python NCCL_DEBUG=INFO train.py ...
```
Comment (awaelchli) [inline docs/source/clouds/cluster.rst:303]:
```suggestion
    NCCL_DEBUG=INFO python train.py ...
```

File: docs/source/clouds/cluster.rst [hunk 2]
```diff
@@ -301,26 +301,3 @@ Set the ``NCCL_DEBUG=INFO`` environment variable to see the ACTUAL error.
...
-
-
-Distributed sampler
--------------------
-
-Normally now you would need to add a
-:class:`~torch.utils.data.distributed.DistributedSampler` to your dataset, however
-Lightning automates this for you. But if you still need to set a sampler set the Trainer flag
```
Comment (awaelchli) [inline docs/source/clouds/cluster.rst:311]: Maybe want to refer to the section replace_sampler_ddp with a short section? Like you did in the multi-gpu page?

**Review by awaelchli — APPROVED** [review]

File: docs/source/clouds/cluster.rst (hunk 2 above)
Comment (carmocca) [reply docs/source/clouds/cluster.rst:311]: I chose to remove this entirely because you can probably expect that anybody looking at the multi node section had a look at the multi-gpu section.

Comment (awaelchli) [reply docs/source/clouds/cluster.rst:311]: then let's remove the entire section 4

File: docs/source/common/lightning_module.rst [hunk 3]
```diff
@@ -54,7 +54,7 @@ Notice a few things.
         new_x = torch.Tensor(2, 3)
         new_x = new_x.type_as(x)
 
-5.  There are no samplers for distributed, Lightning also does this for you.
+5.  Lightning handles the distributed sampler for you.
```
Comment (ananthsub) [inline docs/source/common/lightning_module.rst:57]:
```suggestion
5.  Lightning by default handles the distributed sampler for you.
```

just to make clear that people can opt-out if they choose?

**Review by ananthsub — APPROVED** [review]

**Review by tchaton — APPROVED** [review]
LGTM !

File: docs/source/clouds/cluster.rst (hunk 2 above)
Comment (carmocca) [reply docs/source/clouds/cluster.rst:311]: But the NCCL_DEBUG tip is not in multi_gpu.rst. Do you want me to move it?

Comment (awaelchli) [reply docs/source/clouds/cluster.rst:311]: yes but a whole section with Tips and then only the NCCL debug flag is a bit strange.
